src/commands/reword.rs

```diff
@@ -0,0 +1,358 @@
+use crate::core::eventlog::{EventLogDb, EventReplayer};
+use crate::core::formatting::{printable_styled_string, Glyphs};
+use crate::core::node_descriptors::{render_node_descriptors, CommitOidDescriptor, NodeObject};
+use crate::core::rewrite::{
+    execute_rebase_plan, BuildRebasePlanOptions, ExecuteRebasePlanOptions, ExecuteRebasePlanResult,
+    RebasePlanBuilder, RepoResource,
+};
+use crate::git::{CheckOutCommitOptions, Commit, GitRunInfo, MaybeZeroOid, NonZeroOid, Repo};
+
+/// Reword a commit and restack it's descendants.
+#[instrument]
+pub fn reword(
+    effects: &Effects,
+    hashes: Vec<String>,
+    messages: Vec<String>,
+    git_run_info: &GitRunInfo,
+) -> eyre::Result<isize> {
+    let repo = Repo::from_current_dir()?;
+    let references_snapshot = repo.get_references_snapshot()?;
+    let conn = repo.get_db_conn()?;
+    let event_log_db = EventLogDb::new(&conn)?;
+    let event_replayer = EventReplayer::from_event_log_db(effects, &repo, &event_log_db)?;
+    let event_cursor = event_replayer.make_default_cursor();
+    let mut dag = Dag::open_and_sync(
+        effects,
+        &repo,
+        &event_replayer,
+        event_cursor,
+        &references_snapshot,
+    )?;
+
+    let commits = match resolve_commits_from_hashes(&repo, &mut dag, effects, hashes)? {
+        Some(commits) => commits,
+        None => return Ok(1),
+    };
+
+    let messages = match build_messages(&messages, &commits, get_comment_char(&repo)?)? {
+        BuildRewordMessageResult::Succeeded { messages } => messages,
+        BuildRewordMessageResult::IdenticalMessage => {
+            writeln!(
+                effects.get_output_stream(),
+                "The message wasn't edited; nothing to do. Stopping."
+            )?;
+            return Ok(1);
+        }
+        BuildRewordMessageResult::EmptyMessage => {
+            writeln!(
+                effects.get_error_stream(),
+                "Error: the reworded message was empty. Stopping."
+            )?;
+            return Ok(1);
+        }
+    };
+
+    let subtree_roots = find_subtree_roots(&repo, &dag, &commits)?;
+
+    let rebase_plan = {
+        let pool = ThreadPoolBuilder::new().build()?;
+        let repo_pool = RepoResource::new_pool(&repo)?;
+        let mut builder = RebasePlanBuilder::new(&dag);
+
+        for root_commit in subtree_roots {
+            let only_parent_id = root_commit.get_only_parent().map(|parent| parent.get_oid());
+            if only_parent_id == None {
+                eyre::bail!(
+                    "###
+                    Refusing to reword commit {}, which has {} parents.
+                    Rewording only supported for commits with 1 parent.
+                    Stopping; nothing reworded.
+                    ###",
+                    root_commit.get_oid(),
+                    root_commit.get_parents().len(),
+                );
+            }
+            builder.move_subtree(root_commit.get_oid(), only_parent_id.unwrap())?;
```

Comment:
Use `match` instead of `unwrap`, like this:

```suggestion
            let only_parent_id = match only_parent_id {
                Some(only_parent_id) => only_parent_id,
                None => {
                    eyre::bail!(...); // or `return`
                }
            }
            builder.move_subtree(root_commit.get_oid(), only_parent_id)?;
```

Reply:
Done on all points, thank you. I've changed "Stopping; nothing reworded" to "Aborting"; it's more succinct and matches git's own language.